tuna/machine.py

```diff
@@ -283,31 +307,46 @@ def get_properties(self):
 
     return self.cpus, self.gpus
 
-  def write_file(self, contents, filename=None, is_temp=False):
+  def write_file(self, contents: bytes, filename: Union[str, bytes, 'os.PathLike[str]',\
```

Comment:
the filename as an input to this function should only be a string

Reply:
ok.